### PR TITLE
configure replication

### LINKED ISSUE #41: Configure replication of system keyspaces
The operator has initial support for creating a multi-DC cluster across multiple k8s clusters. It needs to configure the replication of system keyspaces, namely system_auth, to include all DCs. This can be done through the management-api by setting system properties as follows,

```
-Dcassandra.system_distributed_replication_dc_names=dc1,dc2,dc3
-Dcassandra.system_distributed_replication_per_dc=3
```

The goal of this ticket is to only address the use case of configuring the replication when creating a new cluster. We will address the scenarios of adding and removing DCs in separate tickets.

┆Issue is synchronized with this [Jiraserver Feature](https://k8ssandra.atlassian.net/browse/K8SSAND-728) by [Unito](https://www.unito.io)
┆Epic: Exposing Cassandra Configuration
┆Issue Number: K8SSAND-728
┆Priority: Medium

### PR DESCRIPTION
**Pull request by jsanda**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #41

**Checklist**

* [ ] Changes manually tested
* [ ] Automated Tests added/updated
* [ ] Documentation added/updated
* [ ] CHANGELOG.md updated (not required for documentation PRs)
* [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

┆Issue is synchronized with this [Jiraserver Task](https://k8ssandra.atlassian.net/browse/K8SSAND-757) by [Unito](https://www.unito.io)
┆Issue Number: K8SSAND-757
┆Priority: Medium

### CODE REVIEW

File: pkg/cassandra/config.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"encoding/json"
+	"github.com/Jeffail/gabs"
+	"strings"
+)
+
+type NodeConfig map[string]interface{}
+
+func getOperatorSuppliedConfig(dcs []string) NodeConfig {
+	return NodeConfig{
+		"jvm-options": NodeConfig{
```
Comment (burmanm) [inline pkg/cassandra/config.go:13]: This won't work with 4.0 ? In the config-builder it uses ``jvm-server-options`` for 4.0 and ``jam-options`` for 3.11.x

Comment (jsanda) [reply pkg/cassandra/config.go:13]: Yeah, I need to add logic for this. The PR is definitely a work in progress :)

File: controllers/k8ssandracluster_controller.go [hunk 2]
```diff
@@ -240,6 +272,30 @@ func (r *K8ssandraClusterReconciler) updateAdditionalSeeds(ctx context.Context,
 	return nil
 }
 
+// addSeedEndpoints returns a new slice with endpoints added to seeds and duplicates
+// removed.
+func addSeedEndpoints(seeds []string, endpoints ...string) []string {
```
Comment (burmanm) [inline controllers/k8ssandracluster_controller.go:277]: hashSet might be easier to use, but this works too.

File: test/kubectl/kubectl.go [hunk 3]
```diff
@@ -103,6 +103,36 @@ func WaitForCondition(condition string, args ...string) error {
 	return err
 }
 
+// Exec executes a command against a Cassandra pod and the cassandra container in
+// particular. This does not currently handle pipes.
+func Exec(opts Options, pod string, args ...string) (string, error) {
```
Comment (burmanm) [inline test/kubectl/kubectl.go:108]: Can be replaced later, but we should avoid exec to shell for these.

**Review by burmanm — APPROVED** [review]

**Comment by jsanda** [comment]
I am closing this since the changes are also in #50, and I don't want to have to go through a messy rebase again where I already did in #50.

**Pull request closed without merging**